### ISSUE TITLE
Add support to read and write yaml file in Ballerina

### ISSUE THREAD

**Issue by daneshk** [ballerina-platform/ballerina-lang#22616]
**Description:**
Add support to read and write properties file in Ballerina, like XML and JSON file support in [1][2]

1. https://ballerina.io/v1-2/learn/by-example/xml-io.html
2. https://ballerina.io/v1-2/learn/by-example/json-io.html

**Comment by BuddhiWathsala** [comment]
@daneshk, we can follow the same approach to implement APIs for the YAML read and write. However, the only doubtful factor is, unlike JSON and XML, ballerina does not have primitive support for YAML. 

Therefore, we might need to use some other data representation such as record, map, or even JSON to represent YAMLs in the ballerina context.

**Comment by daneshk** [comment]
> @daneshk, we can follow the same approach to implement APIs for the YAML read and write. However, the only doubtful factor is, unlike JSON and XML, ballerina does not have primitive support for YAML.
> 
> Therefore, we might need to use some other data representation such as record, map, or even JSON to represent YAMLs in the ballerina context.

Yes, Let's use the same approach for now. We may need to revisit the API soon. Can we use `json` represent yaml data?
